src/BinSkim.Rules/DwarfRules/BA3003.EnableStackProtector.cs

```diff
@@ -114,20 +126,49 @@ static bool analyze(IDwarfBinary binary, out List<DwarfCompileCommandLineInfo> f
                 return !failedList.Any();
             }
 
+            static bool analyzeSymbols(ElfBinary binary, out List<SymbolEntry<ulong>> failedList)
+            {
+                failedList = new List<SymbolEntry<ulong>>();
+
+                var symbols = binary.ELF.Sections.FirstOrDefault(s => s.Type == SectionType.DynamicSymbolTable) as SymbolTable<ulong>;
```

Comment:
>FirstOrDefault [](http://example.com/codeflow?start=50&length=14)

Why do we retrieve/scan only the first section that you retrieve? Does that mean that there is only a single one that meets these criteria?

Does this mean as well that we *always* expect to find one (and therefore enter the foreach that follows?

Reply:
My understanding is that ELFs should only have one dynamic symbol table. I don't actually know how well that assumption holds up in practice. It would be easy enough to simply iterate through any sections that ELFSharp identifies as dynamic symbol tables.